### PR TITLE
Reference theme

### PR DESCRIPTION
**Pull request by kirkkwang**
## ♻️ Fold duplicated theme helpers into the shared modules

a7e71c9c917c39c5ac2446dcf13b69f10a7aa122

Heritage and practice research each carried a member paginator; practice
research also duplicated the viewer check and the license badge. The facet
guard moves onto ThemeHomepage so a new theme inherits it rather than copying it.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎁 Register the reference home and show themes

b036e030f1162b4265c1093678f0a8662c23dc65

The pair is selectable and the Ellison tenant runs it; the views arrive
next. A theme with no wireframe raises AssetNotFound on the appearance
form, so both previews land with the entries rather than after them.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎨 Dress the reference masthead, nav row and footer

b064800f8ad35df0c159b3dde3f1ea638bbe4574

The nav row reserves the search control's height, so the homepage row is
the same height as every other page even though the form only renders off
the homepage. The announcement band unsets Hyrax's red alert, and the
colors come from a --ref-* block on the appearance styles.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎁 Open the reference home page with search and counts

67d2c9b21df302d8d27847c7df0599d188a99dcb

The hero takes its headline from the marketing text and falls back to a
plain one, carries the collections and works counts, and puts the deposit
invitation on the edge. The banner sits under a scrim derived from the
header background, and the eyebrow is lifted off it to clear 4.5:1.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎁 Browse the reference home page by four metadata axes

1e7d5e9c9f45d9c25f8dfe99aacb316d7132a5db

Subject, creator, collection and publisher each read the tenant's own
facet config, so a column is offered only where the facet survives and
the tracks follow however many are left. Asking Solr for one more row
than a column holds is what decides whether the "all" link is worth
showing, and that link opens Blacklight's facet modal in place.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎁 List reference deposits the way scholarship is listed

262c496812f1c493667f6e4ddb97e4be82b07c0e

Each row reads as a citation: object on a mat, serif title, creators cut
at three, then publisher, date and collection with the separators
collapsing around whatever the record is missing. The rail carries the
type, the deposit date off the Solr document, and the rights statement
above the license. Resource type moves to a shared helper, since the
screening room chip was already asking the same question.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎁 Feature reference collections beside featured works

e33e783a1fa51527c25ad6144c3fbf4f065b74ac

Collections read as a titled list with a thumbnail and a work count, works
as a three-up grid, and either half takes the band alone when the other has
nothing to show. Both keep the framework's reorder form, so an editor drags
and saves as they do everywhere else, and object titles across the page now
carry a quiet underline rather than relying on hover alone.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎁 Close the reference home page with the content bands

408419633fe7b9335e5059b646379da146322351

The featured researcher, the about section and the deposit invitation each
sit against a label column, and each renders only when the tenant has
written the content block behind it. The deposit link opens the work type
modal for an editor and routes everyone else through sign in.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎁 Put the object first on the reference record page

531a40a624d73484b81fdc15bc0d55bccc6faa1b

The viewer opens full width, the description follows at reading size, and
metadata, files and child works sit behind three tabs whose links carry the
pane in the query string, so the server can honor it and a tab is
shareable. The rail holds the citation picker and what the work belongs to.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🎨 Carry the reference chrome onto the rest of the pages

ef54c0ba5308a35c5e846a0005d584739ab91372

The catalog, both advanced searches, the content pages and the collection
and file set shows keep hyrax's own views and take the theme's type, color
and controls over them. The facet rail is pinned rather than left on
bootstrap's quarter, the two show pages read like the work show, and the
selected option in a list box is drawn as a gradient, which is the only
background chrome honors there.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>

## 🐛 Reach the pages the reference reskin claimed

a2ae1e3706e57e018c48928b12bb3aee0de4f10d

The record chip was scoped to the homepage block, so every work show
page rendered it as bare text. Catalog metadata was keyed to
dl.document-metadata, which only exists in gallery view; the default
list view emits an unclassed dl inside .metadata.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01Nj4mBL4ns7L5Th2BCUsd2L

## 🎁 Cover the reference theme's untested paths

e722c877a3d4f1bcb751ea1852ddacda21d5857e

The access checks come first: a restricted featured work or collection
must never reach an anonymous home page, and both examples fail when
the filter in ThemeHomepage is removed.

The rest cover branches nothing else reaches — a facet the tenant
deleted, the ?pane= fallback that makes the tab strip work with
scripting off, and the byline and provenance edge cases.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01Nj4mBL4ns7L5Th2BCUsd2L

## 🧹 Drop the color math unit tests

0fe7746f75a2caaae1e8acaaf57369c9d49af5f5

They assert arithmetic against magic hex and luminance constants, so
they break whenever a threshold is tuned and say nothing about what a
page actually renders. The contrast work these guard was found by
measuring served pages, and that is where it stays covered.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01Nj4mBL4ns7L5Th2BCUsd2L

## ♻️ Give practice research one helper module

601d250f3a41a641cf421efdf7a8fcff867d303c

Every other theme keeps its home and show helpers in one module named
after the theme. Practice research was the last pair still split, so
the two merge with their constants and a single private section.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01Nj4mBL4ns7L5Th2BCUsd2L

## 🐛 Contain the catalog cards and the still viewer

95269ef5ac1534d0d5fc60d02cfa761f377a2736

The row separator was hung on #documents, which only exists in the
gallery, masonry and slideshow views, so it drew stray lines across the
cards and never reached the list rows it was written for.

The cards now carry a frame of their own, and gallery_view.scss's
250x300 cover crop gives way to a contained object on a mat. The facet
rail widens to 320px, closer to the other themes.

The still viewer had the same shape of bug: a flex mat at a fixed height
centred content taller than itself, so a work with no IIIF viewer spilled
its page over the record band and the description.

Co-Authored-By: Claude Opus 5 (1M context) <noreply@anthropic.com>
Claude-Session: https://claude.ai/code/session_01Nj4mBL4ns7L5Th2BCUsd2L

### CODE REVIEW

File: app/helpers/theme_helper.rb [hunk 1]
```diff
@@ -25,40 +30,82 @@ def theme_viewer?(presenter)
...
+  def theme_home(presenter_class)
+    @theme_home ||= presenter_class.new(
+      search_service: controller.search_service,
+      response: @response,
+      collections: @collections,
+      featured_work_list: @featured_work_list,
+      featured_collection_list: @featured_collection_list,
+      current_ability:
+    )
+  end
```
[the remark this answers is not in the document]
Comment (kirkkwang) [reply app/helpers/theme_helper.rb:51]: Refuted. Ruby 3.1 added the keyword-argument shorthand, and it resolves the name the same way a bare reference does: local variable if one exists, otherwise a method call on `self`. It does not require a local. Executed on this app's Ruby (3.3.10):

```ruby
class Probe
  def current_ability = :ability_from_method
  def target(current_ability:) = current_ability
  def run = target(current_ability:)
end
Probe.new.run  # => :ability_from_method
```

`current_ability` is a controller/helper method here, so `current_ability:` passes its return value. If this raised `NameError`, `spec/requests/reference_home_theme_spec.rb` could not render the homepage at all — `ref_home` calls `theme_home` on every one of its 9 examples, and they pass.

File: app/views/themes/reference_show/hyrax/base/_part_of_panel.html.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<section class="ref-panel ref-part-of" aria-labelledby="ref-part-of-label">
+  <h2 class="ref-panel-label" id="ref-part-of-label"><%= t('reference.show.part_of.heading') %></h2>
+  <%= render 'relationships', presenter: %>
+</section>
```
[the remark this answers is not in the document]
Comment (kirkkwang) [reply app/views/themes/reference_show/hyrax/base/_part_of_panel.html.erb:4]: Refuted, and this is not new code. `presenter` is a local of this partial, so `presenter:` is the Ruby 3.1 shorthand for `presenter: presenter` — valid, and not a syntax error. Two checks:

- The identical line already ships on `main`: `app/views/themes/heritage_show/hyrax/base/_part_of_panel.html.erb:5` is `<%= render 'relationships', presenter: %>`.
- `spec/requests/reference_show_theme_spec.rb` renders this partial in all 10 of its examples (the rail is on every work show), and the served page returns 200.

A syntax error would fail at template compile time, so nothing on the show page would render.

File: app/views/themes/reference_show/hyrax/base/_metadata.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div class="ref-meta">
+  <dl class="work-show <%= dom_class(presenter) %>" <%= presenter.microdata_type_to_html %>>
+    <%= render 'attribute_rows', presenter: %>
+    <%= presenter.attribute_to_html(:embargo_release_date, render_as: :date, html_dl: true) %>
```
[the remark this answers is not in the document]
Comment (kirkkwang) [reply app/views/themes/reference_show/hyrax/base/_metadata.html.erb:4]: Refuted, same as the `_part_of_panel` thread. `presenter` is a local of this partial, so `presenter:` is the Ruby 3.1 shorthand for `presenter: presenter`.

- `app/views/themes/heritage_show/hyrax/base/_metadata.html.erb:3` on `main` is the same line: `<%= render 'attribute_rows', presenter: %>`.
- `spec/requests/reference_show_theme_spec.rb` asserts against rows this partial renders — `dl.work-show` carries 9 rows with `dom_class` and the microdata type — so the path is covered, and the served page returns 200.

File: app/helpers/theme_helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kirkkwang) [reply app/helpers/theme_helper.rb:51]: Refuted for `theme_home`. The memo is per-request (it lives on the view context), and a request renders exactly one home theme: `ApplicationController#home_page_theme` returns `current_account_site&.home_theme || 'default_home'` — a single string. The four callers are one per theme (`heritage_helper.rb:6`, `practice_research_helper.rb:12`, `screening_room_helper.rb:6`, `reference_helper.rb:6`), each passing only its own presenter class, and no view calls another theme's helper. So two presenter classes cannot reach it in one request; the memo exists to stop the presenter being rebuilt on every partial that needs it.

Same answer for `theme_show_collection` (the suppressed comment on line 70): its only two callers are `reference_show/_work_title.html.erb:4` and `screening_room_show/_work_title.html.erb:4`, each once per page, and a show page renders one work.

File: app/helpers/theme_helper.rb [hunk 4]
```diff
@@ -25,40 +30,82 @@ def theme_viewer?(presenter)
...
 
-    55.step(5, -5) do |percent|
-      lifted = hex.to_s.delete('#').scan(/../).map do |pair|
-        ((pair.to_i(16) * percent) + (255 * (100 - percent))) / 100
-      end
-      return percent if theme_luminance(format('#%02x%02x%02x', *lifted)) >= 0.31
-    end
+  def theme_file_set_ids(presenter, per: MEMBER_ROWS)
+    @theme_file_set_ids ||= paginate_members(presenter.authorized_file_set_ids, :files_page, per:)
   end
```
[the remark this answers is not in the document]
Comment (kirkkwang) [reply app/helpers/theme_helper.rb:55]: Refuted for both paginators — `theme_file_set_ids` here and `theme_child_work_ids` (the suppressed comment on line 59). Within a request every caller passes the same presenter and the same `per:`:

- `reference_show/show.html.erb` calls `ref_show_panes(@presenter)` at line 11 and renders `'files_pane', presenter: @presenter` (78) and `'children_pane', presenter: @presenter` (80) — one object, so the second call is the cache hit the memo is for, avoiding a repeat Solr fetch for the tab count.
- `heritage_show/show.html.erb:48-49` does the same with `@presenter`.
- `practice_research_helper.rb:73,77` pass `per: pr_rows`, which reads `params[:rows]` once and is constant for the request.

No theme renders a second work's file or child list on one page — `grep` for those partials returns only the `presenter: @presenter` renders above. A `per:` that varied mid-request would be a real bug, so if a future theme needs two lists on one page the memo should be keyed then; today there is no such caller.

File: app/views/themes/reference_show/hyrax/base/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        <% panes.each do |key, label, count| %>
+          <li class="nav-item" role="presentation">
+            <a class="nav-link <%= 'active' if key == active_pane %>"
+               id="ref-tab-<%= key %>"
+               href="?<%= request.query_parameters.merge(pane: key).to_query %>#ref-pane-<%= key %>"
+               data-toggle="tab"
+               data-target="#ref-pane-<%= key %>"
+               role="tab"
+               aria-controls="ref-pane-<%= key %>"
+               aria-selected="<%= key == active_pane %>"><%= label %><% if count %><span class="ref-tab-count"><%= number_with_delimiter(count) %></span><% end %></a>
```
[the remark this answers is not in the document]
Comment (kirkkwang) [reply app/views/themes/reference_show/hyrax/base/show.html.erb:59]: Half right, and the half that matters is refuted. Bootstrap does intercept the click — but Hyrax ships `app/assets/javascripts/hyrax/tabs.js`, which binds every `a[data-toggle="tab"]` and pushes the href into the URL:

```js
$('a[data-toggle="tab"]').on('click', function(e) {
  history.pushState({turbolinks: true}, null, $(this).attr('href'));
});
```

So the `?pane=` lands via `pushState` rather than navigation. Measured in Chrome on the reference show page, JS enabled:

| step | URL | active pane | document requests |
| --- | --- | --- | --- |
| on load | `/concern/generic_works/loc-2018667156` | `ref-pane-metadata` | — |
| click Files | `…?pane=files#ref-pane-files` | `ref-pane-files` | **0** |
| reload | `…?pane=files#ref-pane-files` | `ref-pane-files` (`aria-selected="true"`) | 1 |
| shared link, fresh tab | `…?pane=files#ref-pane-files` | `ref-pane-files` | 1 |

The zero document requests confirm Bootstrap handled the click client-side; the URL still carries the param, so refresh and share both preserve the pane. Scripting off, the href is followed normally and the server honours `params[:pane]` through `theme_active_pane` — `spec/requests/reference_show_theme_spec.rb` covers that path, including the fallbacks for a pane that is not shown and one that does not exist.

**Review by ShanaLMoore — APPROVED** [review]
Approving. The helper consolidation is the good half of this - four themes stop carrying their own copy of the same paginator and deposit target, and both `TODO: dedupe` comments get paid off.

Two things, neither blocking:

`theme_brand_mix` and `theme_brand_mix_on` fall back in opposite directions. The second returns 5 explicitly when nothing clears the floor; the first has no fallback, so it returns 55 by way of `Integer#step` returning its receiver - the least-lifted, least-legible end. Worth making explicit either way.

That's also the sort of thing the deleted color math specs would have caught. I take the point that asserting luminance constants breaks on every tuning, but `ThemeColorHelper` now has no coverage at all, and the served-page measurement isn't in the suite. A property-style spec survives tuning: for a handful of brand colors, assert `theme_contrast(theme_lift(hex, theme_brand_mix_on(hex, ground)), ground) >= 4.5`. That pins the invariant rather than the number.

Smaller: 41 `!important` in the new SCSS. Several are annotated as beating Bootstrap's or Hyrax's own, which is the case where specificity can't win, but about half carry no note - `border-radius: 0 !important`, `margin-top: 22px !important`. Worth a pass.

Not a code note: since the shared helpers now back heritage, practice research and screening room too, worth clicking the deposit button and license badge on all three before this merges, not just reference.
